File: python/src/aegis/resources/multimodal.py
"""Multimodal resource — V3 multimodal security endpoints."""

from __future__ import annotations

from typing import Any, Dict, List, Optional

from aegis.resources._base import AsyncResource, SyncResource


class SyncMultimodal(SyncResource):

    def scan(
        self,
        content: Optional[str] = None,
        *,
        image: Optional[Dict[str, Any]] = None,
        audio_transcription: Optional[str] = None,
        check_types: Optional[List[str]] = None,
        **kwargs: Any,
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {}
        if content is not None:
            body["text"] = content
        if image is not None:
            body["image"] = image
        if audio_transcription is not None:
            body["audio_transcription"] = audio_transcription
        if check_types is not None:
            body["check_types"] = check_types
        body.update(kwargs)
        return self._post("/v3/multimodal/scan", json=body)

    def image_attack(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/multimodal/image", json=kwargs)

    def viscra(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/multimodal/viscra", json=kwargs)

    def mml(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/multimodal/mml", json=kwargs)


class AsyncMultimodal(AsyncResource):

    async def scan(
        self,
        content: Optional[str] = None,
        *,
        image: Optional[Dict[str, Any]] = None,
        audio_transcription: Optional[str] = None,
        check_types: Optional[List[str]] = None,
        **kwargs: Any,
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {}
        if content is not None:
            body["text"] = content
        if image is not None:
            body["image"] = image
        if audio_transcription is not None:
            body["audio_transcription"] = audio_transcription
        if check_types is not None:
            body["check_types"] = check_types
        body.update(kwargs)
        return await self._post("/v3/multimodal/scan", json=body)

    async def image_attack(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/multimodal/image", json=kwargs)

    async def viscra(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/multimodal/viscra", json=kwargs)

    async def mml(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/multimodal/mml", json=kwargs)

File: go/multimodal.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// MultimodalService — V3 multimodal security.
type MultimodalService struct {
	t *internal.Transport
}

func (s *MultimodalService) Scan(ctx context.Context, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/multimodal/scan", map[string]any{"text": text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MultimodalService) ScanWith(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/multimodal/scan", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MultimodalService) Image(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/multimodal/image", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MultimodalService) VisCRA(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/multimodal/viscra", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MultimodalService) MML(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/multimodal/mml", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}
